wave_lang/kernel/wave/codegen/read_write.py

```diff
@@ -408,6 +409,25 @@ def _linearize_memref(
     )
 
 
+def _linearize_shared_mem(memory: CustomOp):
```

Comment:
This won't work with the dynamic symbols. Can you also add a test for the dynamic dim?